api/src/main/java/javax/enterprise/context/spi/ManagedContext.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.context.spi;
+
+import javax.enterprise.context.ContextNotActiveException;
+import javax.enterprise.inject.Instance;
+import java.lang.annotation.Annotation;
+
+/**
+ * Representation of a context which has been activated
+ *
+ * @author John D. Ament
+ */
+public interface ManagedContext {
+
+    /**
+     * Provides access to an internal ID used to uniquely identify this active context
+     *
+     * @return the internal ID of this context, if it is supported
+     */
+    String getId();
+
+    /**
+     * Associates the context to the given thread.
+     * @param thread the thread to associate with.  This does not remove associations with other threads, but acts additive
+     * @throws IllegalArgumentException if the given thread is already associated with this context.
+     */
+    void associate(Thread thread);
```

Comment:
I'm not a big fan of explicitly add a dependency to `Thread` in our API.
When re-reading original JSR we wrote on point 3:

> The context model in CDI is based around the thread-bound request-response model. This model is more prevalent in web applications than elsewhere, and, also is less pervasive than it once was. We would like to overhaul the context model to allow the application or container to portably push the active context to CDI when it requires the use of CDI. This change would be largely transparent to applications, and we might want to consider an API facade over it

To be honest, I'm not totally sure of what Pete meant we that, but my feeling is that if we add something context related we should make sure that it could be used in a more modern reactive world.

I'm just wondering if we are not closing doors here.